crates/fuel-streams-core/src/stream/stream_impl.rs

```diff
@@ -199,28 +224,101 @@ impl<S: Streamable> Stream<S> {
         self.store.stream_name.as_str()
     }
 
-    // TODO: This should probably be `subscribe_raw` since it returns pure bytes
+    // Less performant due to our hybrid use of NATS and S3
+    pub async fn subscribe_raw(
+        &self,
+        subscription_config: Option<SubscriptionConfig>,
+    ) -> Result<BoxStream<'_, Vec<u8>>, StreamError> {
+        let mut config = PullConsumerConfig {
+            filter_subjects: vec![S::WILDCARD_LIST[0].to_string()],
+            deliver_policy: DeliverPolicy::All,
+            ack_policy: AckPolicy::None,
+            ..Default::default()
+        };
+
+        if let Some(subscription_config) = subscription_config {
+            config.filter_subjects = subscription_config.filter_subjects;
+            config.deliver_policy = subscription_config.deliver_policy;
+        }
+
+        let config = self.prefix_filter_subjects(config);
+        let consumer = self.store.stream.create_consumer(config).await?;
+
+        Ok(consumer
+            .messages()
+            .await?
+            .then(|message| {
+                let s3_client = Arc::clone(&self.s3_client);
+
+                async move {
+                    let nats_payload = message
+                        .expect("Message must be valid")
```

Comment:
My main concern here (and in other parts of the project as well), is that we are panicking a lot 🙄 IDK, I guess we should handle these errors better to prevent programs from crash entirely

Reply:
Should we let the user handle unexpected errors at this early stage? In this case, we always publish with a value, our program ensures that, so the `expect` there is an assertion that `async_nats` couldn't help us do. The maintainability of our code improves greatly by identifying these sorts of guarantees

Reply:
I understand your perspective, and I see the benefit of using `expect()` to show invariants clearly. However, I worry that depending on panics for minor issues might hurt DX and operational stability. Given that we’re running a streaming service 24/7, even one unexpected value could crash the app, halt the stream, and affect users. This isn't good.

How about we find a middle ground? We could use `expect()` only in critical areas while implementing proper error handling and logging where problems could be eliminated. This way, we keep the guarantees without risking major issues from minor glitches.